src/WinRT.Runtime/Projections.cs

```diff
@@ -25,9 +25,33 @@ namespace WinRT
 #endif
     static class Projections
     {
+        internal enum UiXamlMode
+        {
+            MicrosoftUiXaml,
+            WindowsUiXaml
+        }
+
+        private static UiXamlMode GetUIXamlModeSetting()
+        {
+            if (AppContext.GetData("CsWinRT.UiXamlMode") is string str && Enum.TryParse<UiXamlMode>(str, out var mode))
+            {
+#if !NET5_0_OR_GREATER
+                if (mode == UiXamlMode.WindowsUiXaml)
+                {
+                    throw new NotSupportedException("Windows.UI.Xaml is not supported before .NET 5. Please use built-in WinRT interop for Windows.UI.Xaml experiences");
+                }
+#endif
+                return mode;
+            }
+            return UiXamlMode.MicrosoftUiXaml; // We default to MUX for back-compat with existing projects.
+        }
+
+        internal static UiXamlMode UiXamlModeSetting { get; } = GetUIXamlModeSetting();

```

Comment:
Not needed for the initial prototype, but we might want to have the final version also include an `ILLink.Substitutions.xml` that can swap the body of the property getter with the hardcoded value based on the runtime switch configuration from MSBuild. I mean, it's simple enough but it's nice to have 🙂

Reply:
Yes, that is a good idea! I'll see if I can figure it out (since it's not a bool I'll need to check the right way to do it)